### ISSUE TITLE
Flexible Residues and Ligand Coordinates Located Far Away from Rigid Receptor Coordinates

### ISSUE THREAD

**Issue by ctrolinder** [PKUGaoGroup/DSDPFlex#1]
Hello, 
I have been attempting to dock a ligand library using:

/home/username/Desktop/DSDPFlex_Clone/DSDPFlex/bin/DSDPflex \
--ligand_batch /home/username/Desktop/DSDPFlex_KynLib/KynLib_Gypsum_Ligands/pdbqt_config_text/pdbqt_out_batch_1.txt \
--protein /home/username/Desktop/DSDPFlex_KynLib/Receptor/4kcd_rigid.pdbqt \
--flex /home/username/Desktop/DSDPFlex_KynLib/Receptor/4kcd_flex.pdbqt \
--box_min -0.025 -15.2875 -21.800 \
--box_max 11.225 6.0875 -6.800 \
--ligbox_min -7.943 -6.165 -6.628 \
--ligbox_max 7.085 4.478 7.779 \
--exhaustiveness 384 \
--search_depth 40 \
--top_n 1 \
--log /home/username/Desktop/DSDPFlex_KynLib/Output/Output_log/DSDP_out_batch_1.log

This executes DSDPFlex without any warnings, however, looking at the output files, my flexible receptor coordinates and ligand coordinates after docking are located far from the rigid receptor coordinates. I would greatly appreciate any feedback as to why this may be happening and if there is a way to fix this issue. Here are some abbreviated examples with the rigid and flex Input files first followed by the Output flex and ligand files:

Inputs --

Rigid: 
ATOM      1  N   ASN A   3      -5.631 -10.613 -46.747  1.00 63.07     0.615 N 
ATOM      2  CA  ASN A   3      -4.248 -10.142 -46.747  1.00 62.58     0.408 C 
ATOM      3  C   ASN A   3      -3.885  -9.403 -45.460  1.00 50.69     0.258 C 
ATOM      4  O   ASN A   3      -2.900  -9.745 -44.800  1.00 57.66    -0.271 OA
ATOM      5  CB  ASN A   3      -3.988  -9.243 -47.962  1.00 68.57     0.153 C 
ATOM      6  CG  ASN A   3      -2.629  -8.554 -47.905  1.00 72.27     0.218 C 
ATOM      7  OD1 ASN A   3      -2.542  -7.325 -47.949  1.00 76.13    -0.274 OA
ATOM      8  ND2 ASN A   3      -1.564  -9.344 -47.797  1.00 67.55    -0.370 N 
ATOM      9 1HD2 ASN A   3      -1.663 -10.198 -47.763  1.00 81.03     0.159 HD
ATOM     10 2HD2 ASN A   3      -0.776  -9.001 -47.761  1.00 81.03     0.159 HD

Flex:
BEGIN_RES TYR A  97
REMARK  3 active torsions:
REMARK  status: ('A' for Active; 'I' for Inactive)
REMARK    1  A    between atoms: CA   and  CB  
REMARK    2  A    between atoms: CB   and  CG  
REMARK    3  A    between atoms: CZ   and  OH  
ROOT
ATOM      1  CA  TYR A  97       0.088  -6.677 -15.445  1.00  8.56     0.180 C 
ENDROOT
BRANCH   1   2
ATOM      2  CB  TYR A  97       1.576  -6.456 -15.183  1.00 10.37     0.073 C 
BRANCH   2   3
ATOM      3  CG  TYR A  97       2.043  -5.039 -15.463  1.00  8.72    -0.056 A 
ATOM      4  CD2 TYR A  97       2.121  -4.101 -14.440  1.00  8.38     0.010 A 
ATOM      5  CE2 TYR A  97       2.541  -2.795 -14.697  1.00 10.62     0.037 A 
ATOM      6  CZ  TYR A  97       2.899  -2.439 -15.988  1.00  9.41     0.065 A 
ATOM      7  CE1 TYR A  97       2.837  -3.362 -17.018  1.00 10.53     0.037 A 
ATOM      8  CD1 TYR A  97       2.410  -4.641 -16.756  1.00 10.66     0.010 A 
BRANCH   6   9
ATOM      9  OH  TYR A  97       3.364  -1.173 -16.317  1.00 11.48    -0.361 OA
ATOM     10  HH  TYR A  97       4.036  -0.999 -15.881  1.00 13.74     0.217 HD
ENDBRANCH   6   9
ENDBRANCH   2   3
ENDBRANCH   1   2
END_RES TYR A  97

Outputs --

Flex:
MODEL 0
REMARK DSDP RESULT:      -0.894304
BEGIN_RES TYR A  97
REMARK  3 active torsions:
REMARK  status: ('A' for Active; 'I' for Inactive)
REMARK    1  A    between atoms: CA   and  CB  
REMARK    2  A    between atoms: CB   and  CG  
REMARK    3  A    between atoms: CZ   and  OH  
ROOT
ATOM      1  CA  TYR A  97     -57.745 -60.114 -93.407  1.00  8.56     0.180 C 
ENDROOT
BRANCH   1   2
ATOM      2  CB  TYR A  97     -56.257 -59.893 -93.145  1.00 10.37     0.073 C 
BRANCH   2   3
ATOM      3  CG  TYR A  97     -55.790 -58.476 -93.425  1.00  8.72    -0.056 A 
ATOM      4  CD2 TYR A  97     -55.712 -57.538 -92.402  1.00  8.38     0.010 A 
ATOM      5  CE2 TYR A  97     -55.292 -56.232 -92.659  1.00 10.62     0.037 A 
ATOM      6  CZ  TYR A  97     -54.934 -55.876 -93.950  1.00  9.41     0.065 A 
ATOM      7  CE1 TYR A  97     -54.996 -56.799 -94.980  1.00 10.53     0.037 A 
ATOM      8  CD1 TYR A  97     -55.423 -58.078 -94.718  1.00 10.66     0.010 A 
BRANCH   6   9
ATOM      9  OH  TYR A  97     -54.469 -54.610 -94.279  1.00 11.48    -0.361 OA
ATOM     10  HH  TYR A  97     -53.797 -54.436 -93.843  1.00 13.74     0.217 HD
ENDBRANCH   6   9
ENDBRANCH   2   3
ENDBRANCH   1   2
END_RES TYR A  97

Ligand:
MODEL 0
REMARK DSDP RESULT:      -0.894304
REMARK  3 active torsions:
REMARK  status: ('A' for Active; 'I' for Inactive)
REMARK    1  A    between atoms: C4_4  and  C5_5 
REMARK    2  A    between atoms: C7_10  and  N2_11 
REMARK    3  A    between atoms: C12_18  and  O5_19 
ROOT
HETATM    1  C1  UNL     1     -55.017 -54.195 -80.420  1.00  0.00     0.018 C 
HETATM    2  C2  UNL     1     -55.962 -54.156 -79.672  1.00  0.00    -0.072 C 
HETATM    3  C3  UNL     1     -57.103 -54.103 -78.783  1.00  0.00     0.086 A 
HETATM    4  C4  UNL     1     -57.036 -54.579 -77.453  1.00  0.00     0.155 A 
HETATM    5  N1  UNL     1     -58.141 -54.495 -76.669  1.00  0.00    -0.237 NA
HETATM    6  C6  UNL     1     -59.323 -53.975 -77.088  1.00  0.00     0.103 A 
HETATM    7  C7  UNL     1     -60.430 -53.932 -76.205  1.00  0.00     0.132 A 
HETATM    8  C8  UNL     1     -61.653 -53.425 -76.674  1.00  0.00     0.038 A 
HETATM    9  C9  UNL     1     -61.776 -52.959 -77.980  1.00  0.00     0.003 A 
HETATM   10  C10 UNL     1     -60.686 -52.990 -78.847  1.00  0.00     0.017 A 
HETATM   11  C11 UNL     1     -59.443 -53.498 -78.419  1.00  0.00     0.030 A 
HETATM   12  C12 UNL     1     -58.308 -53.561 -79.267  1.00  0.00     0.099 A 
ENDROOT
BRANCH   4  13
HETATM   13  C5  UNL     1     -55.820 -55.172 -76.845  1.00  0.00     0.230 C 
HETATM   14  O1  UNL     1     -54.748 -55.300 -77.492  1.00  0.00    -0.643 OA
HETATM   15  O2  UNL     1     -55.859 -55.602 -75.524  1.00  0.00    -0.643 OA
ENDBRANCH   4  13
BRANCH  12  16
HETATM   16  O5  UNL     1     -58.330 -53.098 -80.590  1.00  0.00    -0.358 OA
HETATM   17  H5  UNL     1     -59.125 -52.670 -81.041  1.00  0.00     0.217 HD
ENDBRANCH  12  16
BRANCH   7  18
HETATM   18  N2  UNL     1     -60.331 -54.398 -74.841  1.00  0.00    -0.113 N 
HETATM   19  O3  UNL     1     -61.364 -54.643 -74.183  1.00  0.00    -0.532 OA
HETATM   20  O4  UNL     1     -59.112 -54.507 -74.212  1.00  0.00    -0.532 OA
ENDBRANCH   7  18
TORSDOF 3
ENDMDL

**Comment by PKUGaoGroup** [comment]
The cuda may be not well compiled. Please try to re-compile DSDPFlex. Before compiling, change "sm_70" to "sm_60" in ./DSDPFlex_v0.2/Makefile. If this issue is not been solved, please tell me.

**Comment by ctrolinder** [comment]
I attempted to re-compile with that change to Makefile and it appears that I am still having the same issue. Let me know if you need any further information from my end. Thanks!

**Comment by PKUGaoGroup** [comment]
I found that there are some problems of the setting of boxsize in your running script. The x, y, and z of ligbox_min should not be smaller than that of box_min, and the x, y, and z of ligbox_max should not be larger than that of box_max.  It means that the box should cover the ligand box in x, y, and z axis. Please change the setting of boxsize. If the issue is not been solved, please provide all input files (pdbqt files of ligand, rigid_part, and flex_part of proteins) to us using a zip file.

**Comment by ctrolinder** [comment]
Thank you for the quick response. I have made previous attempts at docking without including the ligbox as well and have found that I still encounter the same issues. Perhaps there is something wrong with my preparation of the rigid/flexible pdbqt files. I have used both AutoDock Tools GUI and scripts for preparing the receptor, but either method results in the same docking outputs. I have attached a zip file containing both the input files and output results from docking. I appreciate any input you can give me. Thanks! 
[4kcd_flexdocking.zip](https://github.com/user-attachments/files/17481472/4kcd_flexdocking.zip)

**Comment by PKUGaoGroup** [comment]
I used your inputs to run the flexible docking, and the running script is : 
./DSDPflex --ligand compound.pdbqt --protein receptor_rigid.pdbqt --flex receptor_flex.pdbqt --box_min 28 -30 -36 --box_max 48 -7 -11
there is no problems in the output files, please see the attachment. 
There is no problems of the input pdbqt files. I think the problem happens when the CUDA was not well compiled in DSDPFlex. Please clean all compiled files in DSDPFlex_v0.2 using "make clean" command, and check your computer CUDA version, and change the sm_70 to a suitable one to re-compile DSDPFlex.

**Comment by PKUGaoGroup** [comment]
I used your inputs to run the flexible docking, and the running script is :
./DSDPflex --ligand compound.pdbqt --protein receptor_rigid.pdbqt --flex receptor_flex.pdbqt --box_min 28 -30 -36 --box_max 48 -7 -11
there is no problems in the output files, please see the attachment.
[results.zip](https://github.com/user-attachments/files/17484029/results.zip)

There is no problems of the input pdbqt files. I think the problem happens when the CUDA was not well compiled in DSDPFlex. Please clean all compiled files in DSDPFlex_v0.2 using "make clean" command, and check your computer CUDA version, and change the sm_70 to a suitable one to re-compile DSDPFlex.

**Comment by ctrolinder** [comment]
Thanks for all of your assistance with this! It looks like my original CUDA install when I first tried to compile was problematic. I thought there was still an issue with CUDA even after uninstalling and reinstalling, but after some troubleshooting and finding that my system was able to successfully compile NVIDIA cuda-samples, I decided to clone the repository again and compile DSDPFlex from scratch. Changed my Makefile sm_70 to sm_86 for my gpu (3090 ti) and was able to successfully run my simulations. Everything looks good now. Thanks again!
